src/data/SubmitError.ts

```diff
@@ -0,0 +1,14 @@
+export interface SubmitErrorInterface {
+    code: number;
+    data?: any;
+}
+
+export class SubmitError implements SubmitErrorInterface {
```

Comment:
Shouldn't there be `SubmitValidationError`, with data property and `SubmitError` without `data` property?